### PR TITLE
Fixed the spellcheck suggestions format error from SOLR 5. DHH-814

### PR DESCRIPTION
**Pull request by lsitu**
https://lib-jira.ucsd.edu:8443/browse/DHH-814

### CODE REVIEW

File: spec/controllers/catalog_controller_spec.rb [hunk 1]
```diff
@@ -23,6 +23,7 @@ def assigns_response
     end
   
     it "should have search results for a poor query with multiple key words" do
+      sign_in User.create({:provider => 'developer'})
```
Comment (mcritchlow) [inline spec/controllers/catalog_controller_spec.rb]: why was this `#sign_in` addition needed for this test?

File: app/controllers/catalog_controller.rb [hunk 2]
```diff
@@ -276,7 +276,15 @@ def index
       end
       (@response, @document_list) = get_search_results
 
-	  spelling_words = @response.spelling.words
+      spelling_words = []
+      @response.spelling.words.each do |word|
+        # handle Hash format for solr 5
```
Comment (mcritchlow) [inline app/controllers/catalog_controller.rb]: will this work for Solr 4 and 5, or just 5?

File: spec/controllers/catalog_controller_spec.rb (hunk 1 above)
Comment (lsitu) [inline spec/controllers/catalog_controller_spec.rb]: @mcritchlow  I think the test object is created in curator view and this test is examining the the search result, while the other test won't.

File: app/controllers/catalog_controller.rb (hunk 2 above)
Comment (lsitu) [inline app/controllers/catalog_controller.rb]: It'll work for both SOLR 4 and SOLR 5 as the spellcheck words will be converted to a string array for solr 4 and solr 5.

**Pull request merged**